src/Composer/Util/Auditor.php

```diff
@@ -0,0 +1,252 @@
+    {
+        $this->httpDownloader = $httpDownloader;
+        $this->setFormat($format);
+    }
+
+    /**
+     * Get the format that will be used to output audit results.
+     *
+     * @return string
+     */
+    public function getFormat(): string
+    {
+        return $this->format;
+    }
+
+    /**
+     * Set the format that will be used to output audit results.
+     *
+     * @param string $format
+     * @return self
+     */
+    public function setFormat(string $format): self
+    {
+        if (!in_array($format, [self::FORMAT_TABLE, self::FORMAT_PLAIN])) {
+            throw new InvalidArgumentException('Invalid format.');
+        }
+        $this->format = $format;
+        return $this;
+    }
+
+    /**
+     * @param IOInterface $io
+     * @param BasePackage[] $packages
+     * @param bool $warningOnly If true, outputs a warning. If false, outputs an error.
+     * @return int
+     * @throws InvalidArgumentException If no packages are passed in
+     */
+    public function audit(IOInterface $io, array $packages, bool $warningOnly = true): int
+    {
+        $advisories = $this->getAdvisories($packages);
+        $format = $warningOnly ? 'warning' : 'error';
+        if (count($advisories) > 0) {
+            $numAdvisories = $this->countAdvisories($advisories);
+            $plurality = $numAdvisories === 1 ? 'y' : 'ies';
+            $io->writeError("<$format>Found $numAdvisories security vulnerability advisor$plurality:</$format>");
+            $this->outputAdvisories($io, $advisories);
+            return 1;
+        }
+        $io->writeError('<info>No security vulnerability advisories found</info>');
+        return 0;
+    }
+
+    /**
+     * Get advisories from packagist.org
+     *
+     * @param BasePackage[] $packages
+     * @param ?int $updatedSince Timestamp
+     * @param bool $filterByVersion Filter by the package versions if true
+     * @return string[][][]
+     * @throws InvalidArgumentException If no packages and no updatedSince timestamp are passed in
+     */
+    public function getAdvisories(array $packages = [], int $updatedSince = null, bool $filterByVersion = true): array
+    {
+        if (count($packages) === 0 && $updatedSince === null) {
+            throw new InvalidArgumentException(
+                'At least one package or an $updatedSince timestamp must be passed in.'
+            );
+        }
+
+        if (count($packages) === 0 && $filterByVersion) {
+            return [];
+        }
+
+        // Add updatedSince query to URL if passed in
+        $url = self::API_URL;
+        if ($updatedSince !== null) {
+            $url .= "?updatedSince=$updatedSince";
+        }
+
+        // Get advisories from API
+        $response = $this->httpDownloader->get($url, $this->createPostOptions($packages));
+        $advisories = $response->decodeJson()['advisories'];
+
+        if (count($packages) > 0 && $filterByVersion) {
```

Comment:
```suggestion
        if (count(advisories) > 0 && $filterByVersion) {
```
Oops. I put the wrong variable here. I'll make this change once people have been given a chance to review the or as a whole.

Reply:
@GuySartorelli please append new commits for new changes, it's much easier to do further reviews. I can squash when merging if it turns into a huge mess.

Reply:
Will do